src/lib/services/api-token-service.ts

```diff
@@ -221,8 +232,19 @@ export class ApiTokenService {
 
     public async createApiTokenWithProjects(
         newToken: Omit<IApiTokenCreate, 'secret'>,
-        createdBy: string = SYSTEM_USER.username,
-        createdByUserId: number = SYSTEM_USER.id,
+        createdBy: IApiUser | IUser,
```

Comment:
This is a breaking change on a public method but was just added in 5.8 https://github.com/Unleash/unleash/commit/bfa82d79bf245058612081062345e0ca7c38b7b4

Alternatively, I could do it in a compatible way with this signature and validate that the field is of type string
```suggestion
        createdBy: string | IApiUser | IUser,
```

Reply:
NVM, I had to support this in [869f4ba](https://github.com/Unleash/unleash/pull/5924/commits/869f4ba4cca2ef5b54e00a26ea36de1c18fa4651) due to failing tests. Maybe this can be rewritten but it's also a public method that can be used to extend Unleash.